.github/workflows/documentation-tests.yaml

```
@@ -0,0 +1,75 @@
+name: Documentation build and test
+on:
+  workflow_dispatch:
+    inputs:
+      optional_documentation_branch:
+        description: 'Optional Documentation branch name'
+        required: false
+        type: string
+        default: 'master'
+
+  push:
+    branches:
+      - 'master'
+    paths-ignore:
+      - '**.md'
+      - '**.txt'
+      - '!**/CMakeLists.txt'
+
+  pull_request:
+    branches:
+      - 'master'
+    paths-ignore:
+      - '**.md'
+      - '**.txt'
+      - '!**/CMakeLists.txt'
+
+jobs:
+  ubuntu-build-and-test-documentation:
+    name: Documentation build and test
+
+    if: ${{ !(contains(github.event.pull_request.labels.*.name, 'no-test') ||
+              contains(github.event.pull_request.labels.*.name, 'skip-ci')) }}
+
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Sync eProsima/Fast-DDS repository
+        uses: actions/checkout@v3
+        with:
+          path: src/fastrtps
+
+      - name: Install Fast DDS packages
+        uses: ./src/fastrtps/.github/actions/install-apt-packages
+
+      - name: Install documentation packages
+        uses: ./src/fastrtps/.github/actions/install-apt-packages-documentation
+
+      - name: Download FastDDS and documentation dependencies
+        run: |
+          pip3 install vcstool
+          vcs import src < ./src/fastrtps/.github/workflows/documentation/ci-documentation.repos
+
+      - name: Move to the given Fast DDS Documentation branch
+        run: |
+          set __ACTION_BRANCH_NAME=${{ inputs.optional_documentation_branch }}
+          if [ -z "${BRANCH_NAME}" ]
+          then
+            set __ACTION_BRANCH_NAME=${GITHUB_REF##*/}
+          fi
+          cd src/fastdds-docs
+          if ! git show-ref --quiet refs/heads/${__ACTION_BRANCH_NAME} ; then
+            set __ACTION_BRANCH_NAME="master"
+          fi
+          git checkout ${__ACTION_BRANCH_NAME}
+
+      - name: Create python virtual environment
+        run: |
+          python3 -m venv fastdds-docs-venv
+          source fastdds-docs-venv/bin/activate
+          pip3 install -r src/fastdds-docs/docs/requirements.txt
+
+      - name: Build and test documentation
+        run: |
+          source fastdds-docs-venv/bin/activate
+          colcon build
```

You'd need to add the `--metas` option here
```suggestion
          colcon build --event-handlers=console_direct+
```